src/shared/components/instant-download/instant-download-gene/InstantDownloadGene.tsx

```diff
@@ -215,12 +243,35 @@ const TranscriptSection = (props: TranscriptSectionProps) => {
   );
 };
 
-const hasSelectedOptions = (
-  options: Partial<TranscriptOptions> & { geneSequence: boolean }
-) => {
-  return Object.keys(options).some(
-    (key) => options[key as keyof TranscriptOptions]
+const hasSelectedOptions = (options: Record<string, boolean>[]) => {
+  return options.some((obj) => Object.values(obj).includes(true));
+};
+
+const downloadGeneSequences = async (params: OnDownloadPayload) => {
+  const { geneId } = params;
+
+  const worker = new Worker(
+    new URL(
+      'src/shared/workers/feature-sequence-download/featureSequenceDownload.worker.ts',
+      import.meta.url
+    )
   );
+
+  try {
+    const service = wrap<WorkerApi>(worker);
+    const sequences = await service.downloadSequencesForGene({
+      genomeId: params.genomeId,
+      geneId: params.geneId,
+      geneSequences: params.options.gene,
```

Comment:
Are you planning to rename these keys?

Reply:
Renamed them to `...sequenceTypes` (`geneSequenceTypes`, `transcriptSequenceTypes`).